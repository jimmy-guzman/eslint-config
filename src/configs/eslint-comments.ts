import comments from "@eslint-community/eslint-plugin-eslint-comments/configs";

import { eslintCommentsRules } from "../rules/eslint-comments";

const eslintCommentsConfig = () => {
  return [
    {
      ...comments.recommended,
      name: "jimmy.codes/eslint-comments",
      rules: eslintCommentsRules,
    },
  ];
};

export default eslintCommentsConfig;
