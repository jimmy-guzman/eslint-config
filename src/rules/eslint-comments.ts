import comments from "@eslint-community/eslint-plugin-eslint-comments/configs";

import type { Rules } from "../types";

export const eslintCommentsRules = {
  ...comments.recommended.rules,
  "@eslint-community/eslint-comments/require-description": "error",
} satisfies Rules;
