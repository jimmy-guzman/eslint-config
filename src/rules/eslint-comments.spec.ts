import { eslintCommentsRules } from "./eslint-comments";

test("should create eslintComments rules", () => {
  expect(eslintCommentsRules).toMatchSnapshot();
});
