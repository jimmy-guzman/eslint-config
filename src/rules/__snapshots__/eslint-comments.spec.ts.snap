// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should create eslintComments rules 1`] = `
{
  "@eslint-community/eslint-comments/disable-enable-pair": "error",
  "@eslint-community/eslint-comments/no-aggregating-enable": "error",
  "@eslint-community/eslint-comments/no-duplicate-disable": "error",
  "@eslint-community/eslint-comments/no-unlimited-disable": "error",
  "@eslint-community/eslint-comments/no-unused-enable": "error",
  "@eslint-community/eslint-comments/require-description": "error",
}
`;
